### PR TITLE
BIOIN-1599 scrna qc testing

### PR DESCRIPTION
**Pull request by inbalzelig-ugi**
* Add unit tests for scrna qc pipeline.
* use papermill python package instead of using subprocess with papermill command.

### CODE REVIEW

File: .github/workflows/python-package-conda.yml [hunk 1]
```diff
@@ -56,5 +56,21 @@ jobs:
       - name: Test with pytest
         run: |
           source activate genomics.py3
-          python -m pytest --ignore test/unit/utils/test_db_access.py test/unit
+          python -m pytest --ignore test/unit/utils/test_db_access.py --ignore test/unit/single_cell_qc test/unit
```
Comment (ilyasoifer) [inline .github/workflows/python-package-conda.yml:59]: Can we add 
`from __future__ import annotations`
and have the tests running in the common environment?

**Comment by ilyasoifer** [comment]
@inbalzelig-ugi - could you please fix the broken build?

File: setup/scRNA_qc/poetry.lock
Comment (ilyasoifer) [inline setup/scRNA_qc/poetry.lock:1]: Can I ask what are all these changes and why they are needed?

File: ugvc/pipelines/single_cell_qc/single_cell_qc_pipeline.py
Comment (ilyasoifer) [inline ugvc/pipelines/single_cell_qc/single_cell_qc_pipeline.py:1]: The structure of this file is not according to the conventions of our reports and pipelines. 
In general we try to separate the statistics collection (single_cell_qc_pipeline.py) and generating the report that should do as little calculations as possible. 

Also - would be nice to have the docstrings formatted in numpy style

**Review by ilyasoifer — COMMENTED** [review]
Looks good!

File: .github/workflows/python-package-conda.yml (hunk 1 above)
Comment (inbalzelig-ugi) [reply .github/workflows/python-package-conda.yml:59]: I'm not sure if all the dependencies are installed in the conda env, additionally, I would like it to run with the specific poetry env to test it as well.

File: setup/scRNA_qc/poetry.lock
Comment (inbalzelig-ugi) [reply setup/scRNA_qc/poetry.lock:1]: We use poetry.lock file to define a specific version of a package. See here for more details: https://python-poetry.org/docs/basic-usage/#committing-your-poetrylock-file-to-version-control

File: ugvc/pipelines/single_cell_qc/single_cell_qc_pipeline.py
Comment (inbalzelig-ugi) [reply ugvc/pipelines/single_cell_qc/single_cell_qc_pipeline.py:1]: I'm not sure I follow - do you suggest moving the report generation part to a different file? 
The docstring format changed.

File: .github/workflows/python-package-conda.yml (hunk 1 above)
Comment (ilyasoifer) [reply .github/workflows/python-package-conda.yml:59]: I tried this locally and it seems to me that all dependencies are installed in conda env (not that surprising, it already contains all the internet :) ). 
And no problem to run the tests also in the poetry env.

File: setup/scRNA_qc/poetry.lock
Comment (ilyasoifer) [reply setup/scRNA_qc/poetry.lock:1]: I am a little bit concerned that we had to make this very significant update for an environment that is a month old. Is this a common occurrence or a one off?

File: ugvc/pipelines/single_cell_qc/single_cell_qc_pipeline.py
Comment (ilyasoifer) [reply ugvc/pipelines/single_cell_qc/single_cell_qc_pipeline.py:1]: More precisely, the report generation is normally a different step in the WDL. So usually there is a script for collecting statistics and a script that just creates an html document that presents these (at least this is the convention that in many cases we are able to enforce)

File: setup/scRNA_qc/poetry.lock
Comment (inbalzelig-ugi) [reply setup/scRNA_qc/poetry.lock:1]: Don't mind this file. Poetry creates it automatically and we just need it as part of the repository in git but I didn't write a single line here.

File: ugvc/pipelines/single_cell_qc/single_cell_qc_pipeline.py
Comment (inbalzelig-ugi) [reply ugvc/pipelines/single_cell_qc/single_cell_qc_pipeline.py:1]: I see, but this is not running as part of a wdl at the moment. If you want we can take this as a different task but let's focus on the current PR, unit testing.

File: .github/workflows/python-package-conda.yml (hunk 1 above)
Comment (ilyasoifer) [reply .github/workflows/python-package-conda.yml:59]: @inbalzelig-ugi - this comment is with you, right?

File: ugvc/pipelines/single_cell_qc/single_cell_qc_pipeline.py
Comment (ilyasoifer) [reply ugvc/pipelines/single_cell_qc/single_cell_qc_pipeline.py:1]: ok, fair enough.

**Review by ilyasoifer — APPROVED** [review]

**Pull request merged**